File: desktop/windows/Localyze/SettingsStore.cs
using Windows.Storage;

namespace Localyze;

/// User settings, persisted in ApplicationData.LocalSettings. No external deps.
public sealed class SettingsStore
{
    public static SettingsStore Instance { get; } = new();

    private readonly ApplicationDataContainer _settings = ApplicationData.Current.LocalSettings;

    private const string KeyWebSearch = "webSearchEnabled";
    private const string KeySearxng   = "searxngUrl";
    private const string KeyCodeExec  = "codeExecEnabled";
    private const string KeyOnboarded = "onboarded";

    public bool WebSearchEnabled
    {
        get => _settings.Values[KeyWebSearch] is bool b && b;
        set => _settings.Values[KeyWebSearch] = value;
    }

    public string SearxngUrl
    {
        get => _settings.Values[KeySearxng] as string ?? "https://searx.be/search";
        set => _settings.Values[KeySearxng] = value;
    }

    /// Whether the `run` tool may execute model-generated python/javascript.
    /// Default false — privacy- and safety-first. Toggle in Settings.
    public bool CodeExecEnabled
    {
        get => _settings.Values[KeyCodeExec] is bool b && b;
        set => _settings.Values[KeyCodeExec] = value;
    }

    /// True once the user has completed the first-run flow (hardware probe + model
    /// fetch). MainWindow uses this to decide whether to mount FirstRunView or
    /// ChatView at launch. Defaults to false so a fresh install always onboards.
    public bool Onboarded
    {
        get => _settings.Values[KeyOnboarded] is bool b && b;
        set => _settings.Values[KeyOnboarded] = value;
    }
}

File: desktop/macos/Sources/Localyze/SettingsStore.swift
import Foundation

/// Persisted user settings. UserDefaults-backed so we don't take any deps.
final class SettingsStore {
    static let shared = SettingsStore()

    private let defaults = UserDefaults.standard
    private enum K {
        static let webSearchEnabled = "localyze.webSearchEnabled"
        static let searxngUrl       = "localyze.searxngUrl"
        static let codeExecEnabled  = "localyze.codeExecEnabled"
        static let onboarded        = "localyze.onboarded"
    }

    var webSearchEnabled: Bool {
        get { defaults.bool(forKey: K.webSearchEnabled) }                 // default false → privacy-first
        set { defaults.set(newValue, forKey: K.webSearchEnabled) }
    }

    var searxngUrl: String {
        get { defaults.string(forKey: K.searxngUrl) ?? "https://search.brave.com/search" }
        set { defaults.set(newValue, forKey: K.searxngUrl) }
    }

    /// Whether the `run` tool may execute model-generated python/javascript.
    /// Default false — safety-first. Toggle in the chat UI / settings drawer.
    var codeExecEnabled: Bool {
        get { defaults.bool(forKey: K.codeExecEnabled) }                  // default false
        set { defaults.set(newValue, forKey: K.codeExecEnabled) }
    }

    /// True once the first-run flow has completed (hardware probe + model
    /// download/import + confirmation). Read on launch by `LocalyzeApp` to
    /// decide whether to show `OnboardingView` or `ChatView`.
    var onboarded: Bool {
        get { defaults.bool(forKey: K.onboarded) }
        set { defaults.set(newValue, forKey: K.onboarded) }
    }
}
